### PR TITLE
Integrate new flat transactions reader with Ledger API

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
- fixes outstanding issue with package id ignored when filtering
- adds --events-page-size CLI option to set the fetch size when streaming transactions

changelog_begin
[Sandbox] The --events-page-size option allows to tweak the fetch size when streaming
transactions.
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/SandboxConfig.scala [hunk 1]
```diff
@@ -46,13 +46,16 @@ final case class SandboxConfig(
...
     metricsReportingInterval: Duration,
+    eventsPageSize: Int,
 )
 
 object SandboxConfig {
   val DefaultPort: Port = Port(6865)
 
   val DefaultMaxInboundMessageSize: Int = 4 * 1024 * 1024
 
+  val DefaultEventsPageSize: Int = 10000
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/SandboxConfig.scala]: `10000` seems quite large imho.

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionsSpec.scala [hunk 2]
```diff
@@ -135,40 +135,13 @@ private[dao] trait JdbcLedgerDaoTransactionsSpec extends OptionValues with Insid
...
             filter = Map(alice -> Set.empty, bob -> Set.empty, charlie -> Set.empty),
-            pageSize = 100,
             verbose = true,
           ))
     } yield {
       comparable(result) should contain theSameElementsInOrderAs comparable(lookups)
     }
   }
 
-  it should "return the same result regardless of the page size" in {
```
Comment (gerolf-da) [inline ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionsSpec.scala]: Was this moved somehwere else?

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/SandboxConfig.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/SandboxConfig.scala]: Sure, what would you think would be a more suitable default? 1000?

File: ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionsSpec.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionsSpec.scala]: Not really, the initialization of `eventsPageSize` has been moved to the data access object, which is initialized by the test suite. I would need a second data access object to test this. I thought of ways in which I could make this happen but then I realized I was basically testing `PaginatingAsyncStream` and `groupContiguous`. Let me know if you're ok to drop this.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/SandboxConfig.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/config/SandboxConfig.scala]: Well, I set it to 1000 in https://github.com/digital-asset/daml/pull/5305/commits/c5099b7aadf381bd236b3cebf806dc55e7aeceb2, let me know if that's ok.

**Pull request merged**